File: 1-easy/problems/2243-calculate-digit-sum-of-a-string/solution.py
class Solution:
    def digitSum(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        while len(s) > k:
            s = "".join(map(str, (sum(map(int, s[i:i+k])) for i in range(0, len(s), k))))
        return s

File: 1-easy/problems/2243-calculate-digit-sum-of-a-string/solution.ts
function digitSum(s: any, k: any): boolean | number | string | any {
  """
  :type s: str
  :type k: int
  :rtype: str
  """
  while ((s).length > k) {
      s = "".join(map(str, (sum(map(int, s[i:i+k])) for i in Array.from({length: 0, (s}, (_, i) => i).length, k))))
  return s;
}

export { digitSum };
